### PR TITLE
bugfix: properly validate :name_service url property when it contains dots (127.0.0.1 / robot.local)

### PR DESCRIPTION
**Pull request by planthaber**
name services were not possible when they were ips or other host names containing dots.

### CODE REVIEW

File: lib/rock/webapp/tasks/api.rb [hunk 1]
```diff
@@ -3,7 +3,7 @@
 module Rock
     module WebApp
         module Tasks    
-            
+
```
Comment (doudou) [inline lib/rock/webapp/tasks/api.rb]: Remove stray spaces

File: lib/rock/webapp/tasks/api.rb [hunk 2]
```diff
@@ -96,27 +98,27 @@ def get_operation(name_service, name, operation_name)
                         end
                         
                     end
-    
+
```
Comment (doudou) [inline lib/rock/webapp/tasks/api.rb]: Stray spaces...

File: lib/rock/webapp/tasks/api.rb [hunk 3]
```diff
@@ -96,27 +98,27 @@ def get_operation(name_service, name, operation_name)
                         end
                         
                     end
-    
+                   
                     desc "Lists information about a given task"
-                    get ':name_service/:name' do
+                    get ':name_service/:name', :requirements => { :name_service => ValidHostnameRegex } do
```
Comment (doudou) [inline lib/rock/webapp/tasks/api.rb]: Use the `x: y` syntax instead of `:x => y`

**Comment by planthaber** [comment]
Further issues?

**Comment by doudou** [comment]
Actually, yes. You should really add unit tests for the regex, especially given how complex it is.

Moreover, does it match the wildcard ? ('*') ?

**Comment by planthaber** [comment]
> Moreover, does it match the wildcard ? ('*') ?

It didn't added tests...

**Comment by doudou** [comment]
The branch now can't be merged. I guess conflicts with the recent merged PRs.

**Comment by planthaber** [comment]
It did, merged them manually and rebased.

**Pull request merged**